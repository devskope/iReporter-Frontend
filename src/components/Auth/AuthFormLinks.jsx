import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const AuthFormLinks = ({ authType }) => {
  const [altText, altLink] =
    authType === 'LOGIN'
      ? ['Need an account?', '/signup']
      : ['Have an account?', '/login'];

  return (
    <div className="auth__links">
      {authType === 'LOGIN' && (
        <p>
          {`Forgot Password? `}
          <a href=".">We can help</a>
        </p>
      )}
      <p>
        {altText}
        <Link to={altLink}> Log In</Link>
      </p>
    </div>
  );
};

AuthFormLinks.propTypes = {
  authType: PropTypes.oneOf(['LOGIN', 'SIGNUP']).isRequired,
};

export default AuthFormLinks;
