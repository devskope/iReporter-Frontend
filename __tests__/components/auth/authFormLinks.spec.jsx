import React from 'react';
import { shallow } from 'enzyme';
import AuthFormLinks from '../../../src/components/Auth/AuthFormLinks';

describe('AuthFormLinks', () => {
  it('should render without crashing', () => {
    const wrapper = shallow(<AuthFormLinks authType="LOGIN" />);

    expect(wrapper.find('.auth__links').exists()).toBe(true);
    expect(wrapper.find('Link')).toHaveLength(1);
  });

  it('should match snapshot', () => {
    const wrapper = shallow(<AuthFormLinks authType="SIGNUP" />);

    expect(wrapper).toMatchSnapshot();
  });
});
