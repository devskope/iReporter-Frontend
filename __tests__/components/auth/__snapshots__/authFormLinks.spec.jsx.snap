// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthFormLinks should match snapshot 1`] = `ShallowWrapper {}`;
